### DISCUSSION TITLE
ArgoCD Internal Architecture

### DISCUSSION THREAD

**Discussion by aliasboink** [argoproj/argo-helm#2351] [Q&A] [answered]
Hello, 

I'm wondering if there's any diagram/documentation that would clearly explain the architecture of ArgoCD.
By this I mean: How does each components communicate with one another (the `argocd-server`, the `argocd-repo-server` accessed by `argocd-server` and `argocd-application-controller`, etc.). 

Is there any clear documentation on this or any way to figure it out clearly?
I've looked at architectural documentation, but it's all a bit higher level and it doesn't go micro-service by micro-service.

Thank you a lot!

**Comment by yu-croco** [comment, marked as answer]
Hi @aliasboink ,
Since argo-helm manages Helm Chart (not Argo CD itself), so I think you can get more help in [upstream](https://github.com/argoproj/argo-cd). 🤔 

> I've looked at architectural documentation, but it's all a bit higher level and it doesn't go micro-service by micro-service.

*I googled but the result is same as your opinion.
*As unofficial doc (JP), https://hiroki-hasegawa.hatenablog.jp/entry/2023/05/02/145115 helped me.

**Reply by aliasboink** [reply]
Hello @yu-croco 

I will quickly write a post upstream regarding this.
Thank you for the reply and the link. I've skimmed through it and it seems rather comprehensive - I will check it out in more detail first thing in the morning.

Once again, thanks for the reply!
